include/eeprom/core/dev_data.hpp

```diff
@@ -0,0 +1,388 @@
+#pragma once
+
+#include "accessor.hpp"
+#include "addresses.hpp"
+#include "can/core/message_core.hpp"
+#include "common/core/bit_utils.hpp"
+#include "task.hpp"
+
+namespace eeprom {
+namespace dev_data {
+
+// set the buffer to the max data length
+using DataBufferType = std::vector<uint8_t>;
+using DataTailType = std::vector<uint8_t>;
+
+struct increase_data_cb_param {
+    void* instance;
+    DataTailType data_to_add;
+};
+
+enum TableAction { READ, WRITE, CREATE, INITALIZE };
+
+struct table_entry_action {
+    uint16_t key;
+    types::address offset;
+    types::data_length len;
+    TableAction action;
+};
+
+// helper class to handle reading writing the data_tail value
+template <task::TaskClient EEPromTaskClient>
+class DevDataTailAccessor
+    : public accessor::EEPromAccessor<EEPromTaskClient, DataTailType,
+                                      addresses::lookup_table_tail_begin,
+                                      addresses::lookup_table_tail_length> {
+    using accessor::EEPromAccessor<
+        EEPromTaskClient, DataTailType, addresses::lookup_table_tail_begin,
+        addresses::lookup_table_tail_length>::EEPromAccessor;
+
+  public:
+    explicit DevDataTailAccessor(EEPromTaskClient& eeprom_client,
+                                 accessor::ReadListener<DataTailType>& listener)
+        : accessor::EEPromAccessor<EEPromTaskClient, DataTailType,
+                                   addresses::lookup_table_tail_begin,
+                                   addresses::lookup_table_tail_length>::
+              EEPromAccessor(eeprom_client, listener) {}
+
+  public:
+    auto increase_data_tail(const DataTailType& data_added) -> void {
+        types::data_length amount_to_read;
+        auto read_addr = addresses::lookup_table_tail_begin;
+        auto bytes_remain = addresses::lookup_table_tail_length;
+        const auto cb_param = new increase_data_cb_param{this, data_added};
```

Comment:
similar to vectors, we definitely should not be using `new` - for a variety of reasons, really. for predictability, in a microcontroller context we want to avoid malloc; and in general from a robustness perspective, we want to use smart pointers. here i think we're best served by packing these structs in a variant and putting them in the message by value.